docs/architecture-decision-records/0017-add-childCount-attribute-dr2-ingest-files.md

```diff
@@ -0,0 +1,33 @@
+# 17. Add childCount attribute to dr2-ingest-files DynamoDB table
+
+**Date:** 2024-06-10
+
+## Context
+
+We have a DynamoDB table where data is queried using a global secondary index (GSI) to find child items of a parent item. Occasionally, the index is not fully populated by the time we read the data, resulting in fewer child items being returned than expected. To address this issue, we need to implement a check to ensure all expected items are retrieved.

```

Comment:
Should we specify that we are talking about assets and files or can this apply to assets and folders too?

Reply:
It applied to both.